tests/test_buffers.py

```diff
@@ -0,0 +1,99 @@
+import gym
+import numpy as np
+import pytest
+import torch as th
+from gym import spaces
+
+from stable_baselines3.common.buffers import DictReplayBuffer, ReplayBuffer
+from stable_baselines3.common.env_util import make_vec_env
+from stable_baselines3.common.type_aliases import DictReplayBufferSamples, ReplayBufferSamples
+from stable_baselines3.common.vec_env import VecNormalize
+
+
+class DummyEnv(gym.Env):
+    """
+    Custom gym environment for testing purposes
+    """
+
+    def __init__(self):
+        self.action_space = spaces.Box(1, 5, (1,))
+        self.observation_space = spaces.Box(1, 5, (1,))
+        self._observations = [1, 2, 3, 4, 5]
+        self._rewards = [1, 2, 3, 4, 5]
+        self._t = 0
+        self._ep_length = 100
+
+    def reset(self):
+        self._t = 0
+        obs = self._observations[0]
+        return obs
+
+    def step(self, action):
+        self._t += 1
+        index = self._t % len(self._observations)
+        obs = self._observations[index]
+        done = self._t >= self._ep_length
+        reward = self._rewards[index]
+        return obs, reward, done, {}
+
+
+class DummyDictEnv(gym.Env):
+    """
+    Custom gym environment for testing purposes
+    """
+
+    def __init__(self):
+        self.action_space = spaces.Box(1, 5, (1,))
+        space = spaces.Box(1, 5, (1,))
+        self.observation_space = spaces.Dict({"observation": space, "achieved_goal": space, "desired_goal": space})
+        self._observations = [1, 2, 3, 4, 5]
+        self._rewards = [1, 2, 3, 4, 5]
+        self._t = 0
+        self._ep_length = 100
+
+    def reset(self):
+        self._t = 0
+        obs = {key: self._observations[0] for key in self.observation_space.spaces.keys()}
+        return obs
+
+    def step(self, action):
+        self._t += 1
+        index = self._t % len(self._observations)
+        obs = {key: self._observations[index] for key in self.observation_space.spaces.keys()}
+        done = self._t >= self._ep_length
+        reward = self._rewards[index]
+        return obs, reward, done, {}
+
+
+@pytest.mark.parametrize("replay_buffer_cls", [ReplayBuffer, DictReplayBuffer])
+def test_normalized_sample(replay_buffer_cls):
```

Comment:
Nitpick, but maybe call this "test_replay_buffer_normalization" for clarity :). Other than that LGTM!

I will let @araffin take a look at this, too. CI seems to be acting up as well...